### DISCUSSION TITLE
Tauri IPC not responding (eg. promise is not being fulfilled/not resolving) only when returning complex values?

### DISCUSSION THREAD

**Discussion by SpikeHD** [tauri-apps/tauri#10322] [Q&A] [answered]
In my project, there is a mix of simple Tauri commands (those that return bools or strings or numbers or whatever) and some that return serialize-able `struct`s or other "complex" data. In all cases that I've tested, commands that return simple values work fine, run fine, and the Promises fulfill just fine.

When returning complex objects (for example, a `HashMap`, or some custom struct), the Promise will indefinitely hang. This also includes the results from Tauri plugins, such as the HTTP plugin. I cannot for the life of me figure out why. Here is an example of some tests:
![image](https://github.com/user-attachments/assets/bf44231b-3e4a-44f9-95ef-c61e3cc7dddf)

Curiously, the CrabNebula Devtools shows the results of these commands just fine, so they aren't hanging at some point during their execution, which leads me to believe something is hanging in the serialization or IPC sending process? Some CN Devtools outputs:

<div width="100%">
  <img width="40%" src="https://github.com/user-attachments/assets/6618a030-cb7f-4680-8a86-f7b60638ecca"/>
  <img width="40%" src="https://github.com/user-attachments/assets/6b80d996-9d22-4ed8-bff3-3876df07bd40"/>
</div>

Some other information:

* My project falls back to the old IPC method automatically, and this is wanted/desired
* This seemed to happen on both Windows and Linux, but my memory is hazy
* My project points to an external site, not one running locally
* Capabilities, enabled plugins, features, etc. can be found [here](https://github.com/SpikeHD/Dorion/tree/next/src-tauri)
* Let me know if you need anything else! :)

**Comment by JonasKruckenberg** [comment]
Hey there! Thanks for reporting this, to me it does seem like an issue with Tauri, not something you're doing wrong. So could you please recreate this as an *Issue* and not a *Discussion*? Thanks!

**Reply by SpikeHD** [reply, marked as answer]
Sure thing! Made a new issue here: #10327

**Reply by JonasKruckenberg** [reply]
Thanks!
